pkg/controller/sparkapplication/controller.go

```diff
@@ -349,14 +349,19 @@ func (c *Controller) getUpdatedAppStatus(app *v1alpha1.SparkApplication) v1alpha
 				}
 			}
 			app.Status.AppState.State = newAppState
-			if app.Status.CompletionTime.IsZero() && !currentDriverState.completionTime.IsZero() {
-				app.Status.CompletionTime = currentDriverState.completionTime
+			if app.Status.TerminationTime.IsZero() && !currentDriverState.completionTime.IsZero() {
+				app.Status.TerminationTime = currentDriverState.completionTime
 			}
 		} else {
 			glog.Warningf("invalid driver state for SparkApplication %s/%s: %v", app.Namespace, app.Name, err)
 		}
 	} else {
 		glog.Warningf("driver not found for SparkApplication: %s/%s", app.Namespace, app.Name)
+		if app.Status.AppState.State == v1alpha1.RunningState && app.Status.TerminationTime.IsZero() {
```

Comment:
So the idea is that this will trigger a re-run if re-run conditions are met, right?

Reply:
Is it possible to miss somehow driver termination and SparkApplication will be in success state?

Reply:
I'm not sure, but setting the app state to `FailingState` and `TerminationTime` may trigger a re-run with certain `RestartPolicy`.

Reply:
And that was my intention. I like to handle case where spark streaming app is running and driver disappears

Reply:
Makes sense to me.